src/fraction/approx.rs

```diff
@@ -0,0 +1,337 @@
+    // To mutate the numerator and denominator of the ratio we'll take ownership of both and then
+    // put them back when we're done.
+    let (mut numer, mut denom) = std::mem::take(ratio).into();
+
+    if numer.is_even() {
+        numer /= 2;
+    } else {
+        denom *= 2;
+    }
+
+    *ratio = BigRational::new_raw(numer, denom);
+}
+
+/// Adds two `Ratio`s together without reducing the result to simplest form. This is significantly
+/// faster than using the standard addition operator provided by `num`, and may be used as long as
+/// the result does not need to be in simplest form (e.g. within an algorithm which reduces the
+/// output ratio before returning).
+fn add_ratios_raw(lhs: BigRational, rhs: BigRational) -> BigRational {
+    // Don't bother comparing the denominators because it's highly unlikely that they're equal.
+    // Instead, we just go straight to giving the fractions equal denominators.
+    let (mut lhs_numer, lhs_denom) = lhs.into();
+    let (mut rhs_numer, rhs_denom) = rhs.into();
+
+    let common_denom = lhs_denom.lcm(&rhs_denom);
+
+    let lhs_multiplier = &common_denom / &lhs_denom;
+    let rhs_multiplier = &common_denom / &rhs_denom;
+
+    lhs_numer *= lhs_multiplier;
+    rhs_numer *= rhs_multiplier;
+
+    BigRational::new_raw(lhs_numer + rhs_numer, common_denom)
+}
+
+/// Converts a `SqrtApprox` into a `DynaFraction`.
+fn convert_sqrt_output<StackInt>(approx: SqrtApprox, reduce: bool) -> DynaFraction<StackInt>
+where
+    StackInt: Copy + GenericInteger + Into<BigUint> + TryToConvertFrom<BigUint> + From<u8>,
+{
+    match approx {
+        SqrtApprox::Rational(ratio) => DynaFraction::Rational(Sign::Plus, {
+            let (numer, denom) = ratio.into();
+
+            // Safety: Both parts of the ratio are always positive, so `to_biguint` will never
+            // fail.
+            let (numer, denom) = (
+                DynaInt::from(numer.to_biguint().unwrap()),
+                DynaInt::from(denom.to_biguint().unwrap()),
+            );
+
+            if reduce {
+                // `Ratio::new` always returns the simplest form, so we don't need to explicitly
+                // reduce the ratio.
+                Ratio::new(numer, denom)
+            } else {
+                Ratio::new_raw(numer, denom)
+            }
+        }),
+
+        SqrtApprox::PlusInf => DynaFraction::infinity(),
+        SqrtApprox::Zero => DynaFraction::zero(),
+        SqrtApprox::NaN => DynaFraction::nan(),
+    }
+}
+
+impl<T: Clone + Integer + ToBigInt + GenericInteger> GenericFraction<T> {
+    /// Returns an unsimplified rational approximation of the square root of `self`.
+    ///
+    /// If you need the result to be simplified, use `sqrt_with_accuracy` instead.
+    ///
+    /// The square of the resulting value is guaranteed to be equal to `self` to the number of
+    /// decimal places specified by `accuracy`.
+    pub fn sqrt_with_accuracy_raw(&self, accuracy: &SqrtAccuracy) -> SqrtApprox {
```

Comment:
Perhaps, accuracy could accept `Option<&GenericInteger>`?
It could accept a pre-existing BigUint/BigInt as well as a new stack (or lazy_static) value? `None` would be provided by `Default` implementation for `SqrtAccuracy`?